vortex-array/src/array/sparse/mod.rs

```diff
@@ -26,7 +26,7 @@ pub struct SparseMetadata {
 }
 
 impl SparseArray {
-    pub fn new(indices: Array, values: Array, len: usize, fill_value: Scalar) -> Self {
+    pub fn new_unchecked(indices: Array, values: Array, len: usize, fill_value: Scalar) -> Self {
```

Comment:
we should remove this method

Reply:
And be replaced with try_new().expect() explicitly?

Reply:
yes